samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs

```diff
@@ -21,15 +23,50 @@ protected override async Task<MessagingExtensionActionResponse> OnTeamsMessaging
         {
             switch (action.CommandId)
             {
-                // These commandIds are defined in the Teams App Manifest.
                 case "createCard":
                     return CreateCardCommand(turnContext, action);
-
                 case "shareMessage":
                     return ShareMessageCommand(turnContext, action);
-                default:
-                    return await Task.FromResult(new MessagingExtensionActionResponse());
+                case "webView":
+                    return CreateThumbnailCard(turnContext, action);
+                case "HTML":
+                    return CreateHeroCard(turnContext, action);
+                case "createAdaptiveCard":
+                    return CreateAdaptiveCard(turnContext, action);
+                case "razorView":
+                    return CreateRazorViewCard(turnContext, action);
             }
+            return await Task.FromResult(new MessagingExtensionActionResponse());
+        }
+
+        private MessagingExtensionActionResponse CreateRazorViewCard(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action)
+        {
+            // The user has chosen to create a card by choosing the 'Create Card' context menu command.
+            CreateRazorCardData cardData = JsonConvert.DeserializeObject<CreateRazorCardData>(action.Data.ToString());
+            var card = new HeroCard
+            {
+                Title = "Requested User: " + turnContext.Activity.From.Name,
+                Subtitle = cardData.Title,
+                Text = cardData.DisplayData,
+            };
+
+            var attachments = new List<MessagingExtensionAttachment>();
```

Comment:
Please combine this similar code of creating a `new MessagingExtensionAttachment` into a single method, used by all instances.